Assets/Scripts/Attributes/ModifyHealthAttribute.cs

```diff
@@ -33,5 +35,12 @@ private void OnTriggerEnter2D(Collider2D colliderData)
                 }
             }
         }
+
+        //TODO pull out speed out of FollowPlayer and Player into parent class
+        var speedScriptEnemy = colliderData.gameObject.GetComponent<FollowPlayer>();
+        if (speedScriptEnemy != null)
+        {
+            speedScriptEnemy.ModifySpeed(slowTarget, slowTargetForSeconds);
+        }
     }
```

Comment:
create own script for all onhit effects including damage and use this to reduce/increase Attributes (including health) of gameObject

Reply:
I don't think that an onhit script is necessary, each script which manages an attribute should just have its own onTriggerEnter.